grpc-api/src/main/java/io/deephaven/grpc_api/session/SessionState.java

```diff
@@ -568,7 +574,14 @@ private synchronized void setState(final ExportNotification.State state) {
                 errorHandler = null;
             }
 
-            if (isExportStateTerminal(state)) {
+            if (exportId != NON_EXPORT_ID) {
```

Comment:
I think you're missing a simplifying observation, here: for non-exports, exported is terminal. You don't need to add an extra boolean (`isReleased`) at all. Just always drop on reaching a terminal state with that changed definition.
I think that would enhance the correctness of the check at the start of this method, too.